control/control.go

```diff
@@ -125,6 +128,31 @@ type managesSigning interface {
 	ValidateSignature([]string, string, []byte) error
 }
 
+type metric struct {
```

Comment:
You can reuse metricType (defined [here](https://github.com/intelsdi-x/snap/blob/master/control/metrics.go#L63) in metrics.Go) instead of creating this definition.